### PR TITLE
GAWB-2140: Integrate statsd client into Rawls

### PR DESCRIPTION
**Pull request by rtitle**
https://broadinstitute.atlassian.net/browse/GAWB-2140

This PR does the following:
- Adds [metrics-scala](https://github.com/erikvanoosten/metrics-scala) and [metrics-statsd](https://github.com/ReadyTalk/metrics-statsd) dependencies to Rawls. These pull in Dropwizard Metrics, and indeed are both referenced on the DW [third party libraries](http://metrics.dropwizard.io/3.2.2/manual/third-party.html) page.
- Added `MetricsSpec` to exercise these new libraries and confirm that they work in the expected way (they do).
- Added new metrics config parsing logic to `Boot.scala`. See related firecloud-develop PR: https://github.com/broadinstitute/firecloud-develop/pull/470

**Note**: this PR doesn't actually _do_ anything yet, but it lays the groundwork for adding instrumentation to Rawls. That will be done in subsequent PRs.

-------

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 1]
```diff
@@ -54,6 +59,30 @@ object Boot extends App with LazyLogging {
       slickDataSource.initWithLiquibase(liquibaseChangeLog, changelogParams)
     }
 
+    val metricsConf = conf.getConfig("metrics")
```
Comment (rtitle) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:63]: There aren't unit tests for the config parsing stuff but I verified it manually with a rawls.conf file.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 2]
```diff
@@ -54,6 +59,30 @@ object Boot extends App with LazyLogging {
...
+    val metricsPrefix = {
+      val basePrefix = metricsConf.getString("prefix")
+      if (metricsConf.hasPath("includeHostname") && metricsConf.getBoolean("includeHostname")) {
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala]: `EnhancedScalaConfig` defines some helpers that might be nicer here, like `getBooleanOption`

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 3]
```diff
@@ -54,6 +59,30 @@ object Boot extends App with LazyLogging {
...
+        val hostname = InetAddress.getLocalHost().getHostName()
+        basePrefix + "." + hostname
+      } else {
+        basePrefix
+      }
+    }
+
+    if (metricsConf.hasPath("reporters") && !metricsConf.getIsNull("reporters")) {
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala]: similarly here

File: core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala (hunk 2 above)
Comment (rtitle) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala]: Ooh thanks, I missed that class.

**Review by helgridly — APPROVED** [review]

**Comment by helgridly** [comment]
I really like the commit comment "relax patience".

**Comment by rtitle** [comment]
Thanks, I like my commit comments to have a calming effect :)

**Comment by helgridly** [comment]
😱

**Comment by rtitle** [comment]
git add antidepressant; git commit -m "relax"; git push origin/master --force

😁

**Pull request merged**